doc/OnlineDocs/contributed_packages/trustregion.rst

```diff
@@ -0,0 +1,184 @@
+####################################
+Trust Region Framework Method Solver
+####################################
+
+The Trust Region Framework (TRF) method solver allows users to solve hybrid
+glass box/black box optimization problems in which parts of the system are
+modeled with open, equation-based models and parts of the system are black
+boxes. This method utilizes surrogate models that substitute high-fidelity
+models with low-fidelity basis functions, thus avoiding the direct implementation
+of the large, computationally expensive high-fidelity models. This is done
+iteratively, resulting in fewer calls to the computationally expensive functions.
+
+The TRF module in Pyomo was originally developed by John Eason to implement
+his and Lorenz Biegler's 2018 algorithm [`Eason & Biegler, 2018`_]. The module
+was updated in late 2021 by **Miranda Mundt** in collaboration with
+**John D. Siirola** of Sandia National Labs to implement a newer version of the 
+algorithm as developed by Yoshio and Biegler [`Yoshio & Biegler, 2021`_].
```

Comment:
In general, I try to avoid attribution of code, especially to core developers (the commit log / git blame handles that).  I would prefer something closer to:

> This module implements the method from Yoshio and Biegler [`Yoshio & Biegler, 2021`_], and represents a rewrite of the original 2018 implementation of the algorithm from Eason & Biegler [`Eason & Biegler, 2018`_].